lib/jekyll/url.rb

```diff
@@ -145,7 +145,8 @@ def self.escape_path(path)
       #   pct-encoded   = "%" HEXDIG HEXDIG
       #   sub-delims    = "!" / "$" / "&" / "'" / "(" / ")"
       #                 / "*" / "+" / "," / ";" / "="
-      URI.escape(path, %r{[^a-zA-Z\d\-._~!$&'()*+,;=:@\/]}).encode("utf-8")
+      path = Addressable::URI.encode(path)
+      path.encode("utf-8").sub("#", "%23")
```

Comment:
Aside from escaping `#` (which I cannot understand), Addressable seems to perfectly fit our needs :+1: